endpoints/s3/src/putobject.cpp

```diff
@@ -717,10 +1020,16 @@ void manually_parse_chunked_body_write_to_irods_in_background(
 
 			// if we are done return ok
 			if (current_state == parsing_state::parsing_done) {
-				if (upload_part) {
+				// this is to force that these are destructed in the correct order
+				auto conn_ptr = conn;
+				auto transport_ptr = tp;
+				auto dstream_ptr = d;
+
+				if (ofs->is_open()) {
 					ofs->close();
 				}
-				else {
+				if (!keep_dstream_open_flag && d->is_open()) {
+					logging::debug("{}:{} Closing iRODS file.", __func__, __LINE__);
```

Comment:
i think this is correct...

```suggestion
					logging::debug("{}:{} Closing iRODS data object.", __func__, __LINE__);
```

Reply:
Changed all reference from "iRODS file" to "iRODS data object" in all files.  I also changed all open/close messages to trace instead of debug.